chunks/chunks.go

```diff
@@ -329,22 +332,30 @@ func NewDirReader(dir string, pool chunkenc.Pool) (*Reader, error) {
 
 	var bs []ByteSlice
 	var cs []io.Closer
+	var sz int64 = 0
 
 	for _, fn := range files {
+		fInfo, err := os.Stat(fn)
+		if err != nil {
+			return nil, errors.Wrapf(err, "stat file")
+		}
+		sz += fInfo.Size()
 		f, err := fileutil.OpenMmapFile(fn)
 		if err != nil {
 			return nil, errors.Wrapf(err, "mmap files")
 		}
 		cs = append(cs, f)
 		bs = append(bs, realByteSlice(f.Bytes()))
 	}
-	return newReader(bs, cs, pool)
+	return newReader(bs, cs, pool, sz)
 }
 
 func (s *Reader) Close() error {
 	return closeAll(s.cs...)
 }
 
+func (s *Reader) Size() int64 { return s.size }
```

Comment:
Is that better though, recalculating the size every time that it gets called? The block size should never change so it seemed more efficient to just store it.

Reply:
even if we have hundreds of blocks this would mean only few cpu cycles so wouldn't cause any overhead. Making the code a lot more clear to follow is a good trade off. 
After all golang is all about simplicity :smiley_cat:

Reply:
I wouldn't mind to hear another  opinion, but I really prefer simplicity over saving few CPU cycles.

some caching can be added if it shows some performance hit, but a first glance I doubt it will need any.

Reply:
Sorry for the late response, I'm fine with doing it inside of the Size() method, I'm not super partial to either approach. I feel it's a small difference in understandability/simplicitiy, but if it's better practice then I'm all for it.

Reply:
lets have a look how it will looks and will try to simplify more if possible.